.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,43 @@
+name: Unit test on push and PR
+
+
+on: [push, pull_request]
```

Comment:
On second thought, @CFlaniganMide what do you think about adding `workflow_dispatch` as an event trigger? e.g., for #82 I kind of want to be able to run tests on the initial "added tests" commit, so it'd be nice to be able to trigger this manually (see [the docs](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) for more info)

Reply:
Yeah, I don't see any reason not to add manual dispatches.

Reply:
cool. do you want to add it or should I?

Reply:
I'll add it.

Reply:
added in 65eada6d40214121d7762dc2aa6dc627a5ae7757